packages/in_app_purchase/in_app_purchase/README.md

```diff
@@ -245,6 +246,63 @@ InAppPurchase.instance
     .buyNonConsumable(purchaseParam: purchaseParam);
 ```
 
+### Accessing platform specific product or purchase properties
+
+The function `_inAppPurchase.queryProductDetails(productIds);` will give you a `ProductDetailsResponse` with a list of 
+purchasable products of type `List<ProductDetails>`. This `ProductDetails` class is a platform independent class 
+with only properties available on both platforms. However, when you want to access some specific
+platform properties this is also possible. The `ProductDetails` object is of subtype `GooglePlayProductDetails`
+when the platform is Android and `AppStoreProductDetails` on iOS. Accessing the skuDetails, resp. skProduct gives
+you all the information that is available in the original platform objects.
+
+This is an example on how to get the `introductoryPricePeriod` on Android:
+```dart
+import 'package:in_app_purchase_android/in_app_purchase_android.dart';
+
+if (productDetails is GooglePlayProductDetails) {
+  var skuDetails = (productDetails as GooglePlayProductDetails).skuDetails;
```

Comment:
I thought about removing the cast, but that doesn't work if it is a property of the class. By doing this cast at least it always compiles and otherwise the reader might be confused if it does not. An ugly example of what would not work:
```
  late PurchaseDetails purchaseDetails;
  void deliverProduct(PurchaseDetails details) async {
    purchaseDetails = details;

    if (purchaseDetails is GooglePlayPurchaseDetails) {
      var billingClientPurchase = purchaseDetails.billingClientPurchase;
      print(billingClientPurchase.originalJson);
    }
}
```

Reply:
Understood, in that case leaving the cast would be the right solution.